packages/@react-aria/interactions/src/usePress.ts

```diff
@@ -666,6 +670,10 @@ function shouldPreventDefault(target: Element) {
   return !target.closest('[draggable="true"]');
 }
 
+function shouldPreventDefaultKeyboard(target: Element) {
+  return target.getAttribute('type') !== 'submit';
```

Comment:
if we go with this one we should check that the element is actually a `<button>` or `<input>` element.